src/app/Components/sign-up/sign-up.component.ts

```diff
@@ -11,14 +11,39 @@ import { IUser, CognitoService } from '../../Services/cognito.service';
 export class SignUpComponent {
   loading: boolean;
   user: IUser;
+  public errorInSignUp: string = '';
+  public errorMessage: string = '';
 
   constructor(private router: Router, private cognitoService: CognitoService) {
     this.loading = false;
     this.user = {} as IUser;
   }
 
-  public signUp(): void {
-    this.loading = true;
+  passwordValidate(password: string): void {
+    const regexPassword = /^[\S]+.*[\S]+$/;
+    let validatePassword = (<HTMLInputElement>(
+      document.getElementById('confirmPassword')
+    )).value;
+
+    if (!regexPassword.test(password) && password.length > 5) {
+      this.user.password = '';
+      this.errorMessage =
+        'Password must contain at least 1 uppercase letter and one lowercase letter';
+    }
+    if (password !== validatePassword) {
+      this.user.password = '';
+      (<HTMLInputElement>document.getElementById('confirmPassword')).value = '';
+      this.errorMessage = 'The passwords do not match';
+    }
+  }
+  seeError(e: Error): void {
+    this.errorInSignUp = 'There was a problem';
+    this.loading = false;
```

Comment:
it would be better to keep `this.loading = false` in the same method where it was declared true